src/analyses/goto_check.cpp

```diff
@@ -1856,24 +1866,47 @@ optionalt<exprt> goto_checkt::rw_ok_check(exprt expr)
 class flag_resett
 {
 public:
+  flag_resett(messaget &_log, bool _warn_on_seen = false)
+    : log(_log), warn_on_seen(_warn_on_seen)
+  {
+  }
+
   /// Store the current value of \p flag and then set its value to \p new_value.
   void set_flag(bool &flag, bool new_value)
   {
+    if(warn_on_seen && seen(flag))
+    {
+      log.warning() << "Found more than one CPROVER enable/disable check "
+                       "pragma on a same goto instruction"
+                    << messaget::eom;
```

Comment:
I think the error message could be improved a bit. Having "more than one CPROVER enable/disable check" is fine as long as they are _different_ checks.
I think we want to say that we _the same check_ being enabled/disabled multiple times times at a goto instruction.

Reply:
message updated to include the name of the check